packages/back-end/src/jobs/updateExposureQueriesDimensionSlices.ts

```diff
@@ -0,0 +1,266 @@
+        if (filteredExposureQueries.length === 0) {
+          return null;
+        }
+
+        return {
+          organizationId: dataSource.organization,
+          dataSourceId: dataSource.id,
+          exposureQueryIds: filteredExposureQueries.map((eq) => eq.id),
+        };
+      })
+      .filter((x) => x !== null);
+
+    return dataSourcesAndExposureQueryIdsMap;
+  } catch (e) {
+    // Silently fail as this is not critical -- but log it so we can fix it
+    logger.error(e, "Failed to queue dimension slices updates");
+    return [];
+  }
+}
+
+export const updateSingleDatasourceExposureQueriesDimensionSlices = async (
+  job: Job<{
+    organizationId: string;
+    dataSourceId: string;
+    exposureQueryIds: string[];
+  }>,
+) => {
+  const organizationId = job.attrs.data?.organizationId;
+  const dataSourceId = job.attrs.data?.dataSourceId;
+  const exposureQueryIds = job.attrs.data?.exposureQueryIds;
+
+  if (
+    !organizationId ||
+    !dataSourceId ||
+    !exposureQueryIds ||
+    !exposureQueryIds.length
+  ) {
+    logger.warn(
+      {
+        organizationId,
+        dataSourceId,
+        exposureQueryIds,
+      },
+      "Invalid job parameters for dimension slices update",
+    );
+    return;
+  }
+
+  const context = await getContextForAgendaJobByOrgId(organizationId);
+  const dataSource = await getDataSourceById(context, dataSourceId);
+  if (!dataSource) {
+    logger.warn(
+      `Data source ${dataSourceId} not found for dimension slices update`,
+      {
+        organizationId,
+        dataSourceId,
+      },
+    );
+    return;
+  }
+
+  const integration = await getIntegrationFromDatasourceId(
+    context,
+    dataSourceId,
+    true,
+  );
+
+  // We need to keep track of all queries, even the ones we are not changing, so we can update the data source
+  const allExposureQueries = new Map(
+    dataSource.settings.queries?.exposure?.map((q) => [q.id, { ...q }]),
+  );
+
+  for (const queryId of exposureQueryIds) {
+    const exposureQuery = allExposureQueries.get(queryId);
+    if (!exposureQuery) {
+      logger.warn(
+        {
+          organizationId,
+          dataSourceId,
+          exposureQueryId: queryId,
+        },
+        `Exposure query not found in data source`,
+      );
+      continue;
+    }
+
+    const model = await createDimensionSlices({
+      organization: organizationId,
+      dataSourceId,
+      queryId,
+    });
+
+    const queryRunner = new DimensionSlicesQueryRunner(
+      context,
+      model,
+      integration,
+    );
+
+    queryRunner.startAnalysis({
+      exposureQueryId: queryId,
+      lookbackDays: DEFAULT_LOOKBACK_DAYS,
+    });
+
+    await queryRunner.waitForResults();
+    const outputModel = queryRunner.model;
+
+    exposureQuery.dimensionSlicesId = outputModel.id;
+    exposureQuery.dimensionMetadata = exposureQuery.dimensions.map((d) => ({
+      dimension: d,
+      specifiedSlices:
```

Comment:
If the output has no slices or errors, we should consider not updating the results at all. I would rather we preserve stale data than overwrite it with nothing.